landing_page/spec/controllers/user_controller_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+
+    context "create a new user account with acceptable credentials" do
+
+        it "successfully adds a new user" do
+            expect{create_new_user}.to change(User, :count).by(1)
+        end
+
+        it "logs the user in upon account creation" do
+            create_new_user
+            expect(session[:user_id]).to_not be_nil
+        end
+
+        it "redirects a new user to their profile page upon creation" do
+            create_new_user
+            expect(response).to redirect_to(assigns(:user))
+        end
+
+    end
+
+    context "GET #index" do
+
+        it "shows the list of users, expects a redirect response to be 200" do
+            get :index
+            expect(response.status).to eq(200)
+        end
+
+    end
+
+    context "GET #show" do
+
+        it "shows a user their profile, expect redirect response to be 200" do
+            user = FactoryBot.create :user
+            user.save
+            login(user)
+
+            get :show, params: { id: user.id }
+            expect(response.status).to eq(200)
```

Comment:
Same as my previous comment. What if the page is blank but it returns a `200`?

Reply:
The page shouldn't be blank. In order to access this page, a user needs to be logged in, so their data will have been saved  and this is what they're accessing. I'll look into a better way of verifying this than just a 200 response though

Reply:
Consider the proper scenario, "shows a user their profile", how can we assert that the provided user is seeing their profile, and not another user's?

Also, you wrote:
> The page shouldn't be blank.

That's your assumption based on the code, or in other words, that's the code intention. The role of a test is to challenge that assumption by explicitly testing the resulting data.

Reply:
Resolved in commit fa0508cf969ee02dc88928e9c412d18df6ea8a74